_spinnaker/terraform_integration.md

```diff
@@ -4,114 +4,113 @@ title: Terraform Integration
 order: 141
 ---
 
-The following tutorial will walk you through how to setup our Terraform integration and execute Terraform code stored in a Github repo as part of a Spinnaker pipeline. We'll assume that you're using Terraform to create and manage infrastructure on AWS.
-
 Note that the Terraform integration is in beta while we work on improving the user experience.
 
+{:.no_toc}
+## Overview
+
+Use Armory Spinnaker's Terraform Integration to integrate your infrastructure-as-code with Terraform workflow into your Spinnaker instance. Manage your application infrastructure as part of a Spinnaker pipeline. The following tutorials walk you through how to setup Armory's Terraform integration and execute Terraform code stored in a Git repo as part of a Spinnaker pipeline. The examples on this page describe a workflow for using Terraform to create and manage infrastructure on AWS. 
+
+{:.no_toc}
+### Under the hood
+
+At the core of the Terraform Integration is the Terraformer service. This service is reponsible for fetching your Terraform projects from source and executing various Terraform commands against them. When a `terraform` stage starts, Orca submits the task to Terraformer and monitors it until completion. Once a task is submitted, Terraformer fetches your target project, runs `terraform init` to initialize the project, and then runs your desired `action` (`plan` or `apply`). If the task is successful, the stage gets marked successful as well. If the task fails, the stage gets marked as a failure and halts the pipeline.
+
```

Comment:
Available actions are `plan`, `apply`, `destroy` or `output`. I'm not sure if it's useful or more confusing to call that out here.